virtcontainers/utils/utils.go

```diff
@@ -232,17 +235,39 @@ func BuildSocketPath(elements ...string) (string, error) {
 	return result, nil
 }
 
+func supportsVsocks() (bool, error) {
+	if _, err := os.Stat(VSockDevicePath); os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	if _, err := os.Stat(VHostVSockDevicePath); os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
 // SupportsVsocks returns true if vsocks are supported, otherwise false
-func SupportsVsocks() bool {
-	if _, err := os.Stat(VSockDevicePath); err != nil {
-		return false
+func SupportsVsocks() (bool, error) {
+	supportVsock, err := supportsVsocks()
+	if err != nil {
+		return false, err
 	}
 
-	if _, err := os.Stat(VHostVSockDevicePath); err != nil {
-		return false
+	// in case vhost_vsock module isn't built-in, but could be loaded
+	if !supportVsock {
+		if output, err := exec.Command("modprobe", "-i", VHostVsockModule).CombinedOutput(); err != nil {
```

Comment:
IMO, we shouldn't trust in the envar `PATH`, the full path to `modprobe` should be used instead

Reply:
ok. I will use `/sbin/modprobe` to instead. ;)

Reply:
updated. plz review. ;)